### PR TITLE
Add device registry to MQTT switch

### PR DESCRIPTION
**Pull request by OttoWinter**
**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#17244

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/switch.mqtt.markdown [hunk 1]
```diff
@@ -103,6 +103,35 @@ value_template:
...
+  description: 'Information about the device this switch is a part of to tie it into the [device registry](https://developers.home-assistant.io/docs/en/device_registry_index.html). Only works through [MQTT discovery](/docs/mqtt/discovery/) and when [`unique_id`](#unique_id) is set.'
+  required: false
+  type: map
+  keys:
+    identifiers:
+      description: 'A list of IDs that uniquely identify the device. For example a serial number.'
+      required: false
+      type: list, string
+    connections:
+      description: 'A list of connections of the device to the outside world as a list of tuples `[connection_type, connection_identifier]`. For example the MAC address of a network interface: `[["mac", "02:5b:26:a8:dc:12"]]`.'
```
Comment (DubhAd) [inline source/_components/switch.mqtt.markdown]: Can you make the same clarity update here please.

**Review by DubhAd — DISMISSED** [review]

File: source/_components/switch.mqtt.markdown [hunk 2]
```diff
@@ -103,6 +103,35 @@ value_template:
...
+  type: map
+  keys:
+    identifiers:
+      description: 'A list of IDs that uniquely identify the device. For example a serial number.'
+      required: false
+      type: list, string
+    connections:
+      description: 'A list of connections of the device to the outside world as a list of tuples `[connection_type, connection_identifier]`. For example the MAC address of a network interface: `"connections": [["mac", "02:5b:26:a8:dc:12"]]`.'
+      required: false
+      type: list, tuple
```
Comment (frenck) [inline source/_components/switch.mqtt.markdown]: Same as the other PR, `tuple` is not a valid type for the documentation.

File: source/_components/switch.mqtt.markdown [hunk 3]
```diff
@@ -103,6 +103,35 @@ value_template:
   description: "Defines a [template](/docs/configuration/templating/#processing-incoming-data) to extract a value from the payload."
   required: false
   type: string
+device:
+  description: 'Information about the device this switch is a part of to tie it into the [device registry](https://developers.home-assistant.io/docs/en/device_registry_index.html). Only works through [MQTT discovery](/docs/mqtt/discovery/) and when [`unique_id`](#unique_id) is set.'
```
Comment (frenck) [inline source/_components/switch.mqtt.markdown:107]: `Information about the device this switch` sounds/reads weird.
Adding comma's would probably help.

**Review by frenck — CHANGES_REQUESTED** [review]
@OttoWinter Left you some comments, could you take a look? 👍

**Review by frenck — APPROVED** [review]

**Pull request merged**